command/server_dev.go

```diff
@@ -0,0 +1,900 @@
+	}
+	options.authAppRole.roleID = roleID
+
+	r = &logical.Request{
+		Operation:   logical.UpdateOperation,
+		ClientToken: init.RootToken,
+		Path:        fmt.Sprintf("auth/approle/role/%s/secret-id", options.authAppRole.roleName),
+	}
+	resp, err = core.HandleRequest(ctx, r)
+	if err != nil {
+		return fmt.Errorf("error retrieving approle id: %w", err)
+	}
+	if resp.IsError() {
+		return fmt.Errorf("failed to retrieve approle id: %w", resp.Error())
+	}
+
+	secretID, ok := resp.Data["secret_id"].(string)
+	if !ok {
+		return fmt.Errorf("failed to parse secret id")
+	}
+
+	options.authAppRole.secretID = secretID
+
+	return nil
+}
+
+func qsBanner(c *ServerCommand, o *quickstartOptions, e string) {
+	if o == nil {
+		// no quickstart options seem to be set
+		return
+	}
+
+	c.UI.Warn(wrapAtLength("A set of quickstart options have been enabled!"))
+	c.UI.Warn("Read More: https://www.vaultproject.io/docs/concepts/dev-server")
+	c.UI.Warn("")
+	c.UI.Warn(wrapAtLength("An identity entity with an associated alias has been created. " +
+		"To explore the restricted account, authenticate with the following information"))
+	c.UI.Warn("Read More About Identity: https://www.vaultproject.io/docs/concepts/identity")
+	c.UI.Warn("Read More About Userpass: https://www.vaultproject.io/docs/auth/userpass")
+	c.UI.Warn("Read More About Policies: https://www.vaultproject.io/docs/concepts/policies")
+	c.UI.Warn("")
+
+	c.UI.Warn(fmt.Sprintf("               Name: %s", o.identity.name))
+	c.UI.Warn(fmt.Sprintf("         Enitity ID: %s", o.identity.entityID))
+	c.UI.Warn(fmt.Sprintf("           Policies: %s", o.identity.policies))
+	c.UI.Warn("Associated Userpass:")
+	c.UI.Warn(fmt.Sprintf("         - Username: %s", o.authUserpass.username))
+	c.UI.Warn(fmt.Sprintf("         - Password: %s", o.authUserpass.password))
+	c.UI.Warn("")
+
+	c.UI.Warn(wrapAtLength("Applications can login with the following AppRole credentials:"))
+	c.UI.Warn("Read More: https://www.vaultproject.io/docs/auth/approle")
+	c.UI.Warn("")
+	c.UI.Warn(fmt.Sprintf("Login Path: %s/v1/auth/approle/login", e))
+	c.UI.Warn(fmt.Sprintf(" Role Name: %s", o.authAppRole.roleName))
+	c.UI.Warn(fmt.Sprintf("   Role ID: %s", o.authAppRole.roleID))
+	c.UI.Warn(fmt.Sprintf(" Secret ID: %s", o.authAppRole.secretID))
+	c.UI.Warn(fmt.Sprintf("  Policies: %s", o.authAppRole.policies))
+	c.UI.Warn("")
+
+	c.UI.Warn(wrapAtLength("Polices have been created to enable the following permissions:"))
+	c.UI.Warn("")
+	for name, policy := range o.policyMap {
+		c.UI.Warn(fmt.Sprintf("Policy Name: %s", name))
+		c.UI.Warn(fmt.Sprintf("Policy Path: %s/v1/sys/policy/%s", e, name))
+		c.UI.Warn(fmt.Sprintf("     Policy: %s", policy))
+		c.UI.Warn("")
+	}
+
+	c.UI.Warn(wrapAtLength("A sample secret is available at:"))
+	c.UI.Warn("")
+	c.UI.Warn(fmt.Sprintf("  Secret Path: %s/v1/secret/data/sample-secret", e))
+	c.UI.Warn(fmt.Sprintf("Metadata Path: %s/v1/secret/metadata/sample-secret", e))
```

Comment:
I wonder if people may say "okay I see the full URL paths here but what do I do with that?"

Perhaps it's worth adding a one-liner like "This secret can be accessed with `vault kv get -mount=secret sample-secret`." ? 

(using that 1.11 mount flag syntax to jump past any /data/ confusion, assuming the dev-quickstart flag isn't going to be backported before 1.11.)

Reply:
There was concern about making the banner too verbose and difficult to read, but I'm happy to add these in. My worry is that the same reasoning could be applied to the sections, for example, including the CLI command for policy operations, which would indeed make the banner a few lines longer. Maybe this is better suited for the docs?